### ISSUE TITLE
Magento 2.4.3 Customer has no any vault  token save, but vault payment method appears in Admin checkout

### ISSUE THREAD

**Issue by jignesh-coder** [magento/magento2#33954]
### Preconditions (*)

1. Magento 2.4.3

### Steps to reproduce (*)

1. Go to Magento Admin panel
2. Setup Braintree payment method and enable vault payment (Stores - Configuration - Sales - Payment Methods - Configure Braintree - Enable Card Payments and Enable Vault for Card )
3. Reindex and Clear cache
4. Admin - Sales - Orders - Create new Order - Create New Customer / or select any existing customer
5. Add Products
Payment Method section - **Issue:** "Stored Cards" Payment option is getting displayed
In payment method section it show token payment
6. Select the option and submit Order: **Issue:** "Order saving error: Public hash should be defined" error message is displayed to the user.
**Note:** There is no issue on Front-end - "Stored Cards" payment option is not displayed for signed in user in Checkout page

### Expected result (*)

1. If customer has no token saved, than Vault Payment should not be shown in Payment section of Admin Order creation page

### Actual result (*)

Order saving error: Public hash should be defined" error message is displayed to the user.
![image](https://user-images.githubusercontent.com/60198231/132174214-30df36d1-4a4c-45cd-a37c-f001660057f5.png)

![image](https://user-images.githubusercontent.com/60198231/132174140-1068ba73-bd35-441b-b103-fda12d9f3d26.png)

Even after saving the Card details - User cannot able to make payment from Admin using Stored Cards option
![image](https://user-images.githubusercontent.com/60198231/133245843-3e39673f-a77d-4df6-8018-6f4925474d6f.png)

There is no issue on front-end - User is able to make payment using stored card and place Order - **No issue**
![image](https://user-images.githubusercontent.com/60198231/133245754-e17bd1b6-6705-411a-afa1-dae8776237e7.png)

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-November** [comment]
Verified the issue on Magento 2.4-develop branch and the issue is reproducible:
**Steps to reproduce:**
Stores - Configuration - Sales - Payment Methods - Configure Braintree - Enable Card Payments and Enable Vault for Card Payments - Yes and Save
Reindex and Clear cache
Admin - Sales - Orders - Create new Order - Create New Customer / or select any existing customer
Add Products
Payment Method section - **Issue: "Stored Cards" Payment option is getting displayed**
Select the option and submit Order: **Issue: "Order saving error: Public hash should be defined" error message is displayed to the user.**
Note: There is no issue on Front-end - "Stored Cards" payment option is not displayed for signed in user in Checkout page
![image](https://user-images.githubusercontent.com/60198231/131657170-f1e30bd3-ab9d-4822-b050-7e6f166ff5f1.png)

![image](https://user-images.githubusercontent.com/60198231/131657086-8c928c29-d26e-4771-af7c-e89375b09766.png)

**Comment by SilinMykola** [comment]
@magento I am working on this

**Comment by jignesh-coder** [comment]
I solved this error from my own way

Add code in di.xml
```
<type name="Magento\Quote\Model\Quote\Payment">
        <plugin name="payment-data" type="Vendor\Extension\Plugin\Payment" />
    </type>
```
    
in file Payment.php add code

<```
?php

namespace Vendor\Extension\Plugin;
class Payment
{
    public function beforeImportData(
        \Magento\Quote\Model\Quote\Payment $subject,
        $data
    ) {
        $vaultMethod = preg_replace('/^([a-z_]+vault)_\d+$/i', '$1', $data['method']);
        if (!is_null($vaultMethod) && $vaultMethod !== $data['method']) {
            $data['method'] = $vaultMethod;
        }
        return [$data];
    }
}
```

**Comment by engcom-Alfa** [comment]
HI @jignesh-coder , @SilinMykola and @yurii-denyshchenko Thanks for your contribution here.
Internal team has picked up the ticket to fix with the internal ticket number `AC-1119`.
Thanks

**Comment by engcom-Hotel** [comment]
Hello @<reporter>,

As I can see this issue got fixed in the scope of the internal Jira ticket AC-1119 by the internal team
Related commits: https://github.com/magento/magento2/commit/fd7e45e945796afcb39ef96c57121f494d651754

Based on the Jira ticket, the target version is 2.4.6.

Thanks

**Comment by atIOCrON** [comment]
We had this issue on Magento ver. 2.4.5-p1 so we upgraded to Magento ver. 2.4.6-p2 and I can confirm it is now resolved.

Unfortunately, the same issue exists if you enable PayPal through Braintree and set Vault Enabled to yes. In this case, "Stored Accounts (PayPal)" will show as a payment method for admin orders even if there is no saved PayPal account.

I've added a new issue to solve it:

https://github.com/magento/magento2/issues/38010
